app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -185,6 +186,9 @@ impl List {
         for (id, entry) in ids_and_entries {
             self.allowed_favorites.insert(id);
             let component = Component::new_from_database_entry(id, entry.clone_ref());
+            if component_is_private(&component) {
+                continue;
```

Comment:
We are preferring not to use `break` nor `continue`. Can you filter the list and iter over it instead?